homeassistant/components/tado/device_tracker.py

```diff
@@ -1,149 +1,132 @@
-import voluptuous as vol
-
-from homeassistant.components.device_tracker import (
-    DOMAIN,
-    PLATFORM_SCHEMA as BASE_PLATFORM_SCHEMA,
-    DeviceScanner,
-)
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
-from homeassistant.core import HomeAssistant
-from homeassistant.helpers.aiohttp_client import async_create_clientsession
-import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.typing import ConfigType
-from homeassistant.util import Throttle
-
-_LOGGER = logging.getLogger(__name__)
-
-CONF_HOME_ID = "home_id"
-
-MIN_TIME_BETWEEN_SCANS = timedelta(seconds=30)
-
-PLATFORM_SCHEMA = BASE_PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_HOME_ID): cv.string,
-    }
-)
-
-
-def get_scanner(hass: HomeAssistant, config: ConfigType) -> TadoDeviceScanner | None:
-    """Return a Tado scanner."""
-    scanner = TadoDeviceScanner(hass, config[DOMAIN])
-    return scanner if scanner.success_init else None
-
-
-Device = namedtuple("Device", ["mac", "name"])
+from homeassistant.components.device_tracker import SourceType
+from homeassistant.components.device_tracker.config_entry import ScannerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
+from .const import DATA, DOMAIN, SIGNAL_TADO_MOBILE_DEVICE_UPDATE_RECEIVED
 
-class TadoDeviceScanner(DeviceScanner):
-    """Scanner for geofenced devices from Tado."""
+_LOGGER = logging.getLogger(__name__)
 
-    def __init__(self, hass, config):
-        """Initialize the scanner."""
-        self.hass = hass
-        self.last_results = []
 
-        self.username = config[CONF_USERNAME]
-        self.password = config[CONF_PASSWORD]
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Tado device scannery entity."""
+    _LOGGER.debug("Setting up Tado device scanner entity")
+    tado = hass.data[DOMAIN][entry.entry_id][DATA]
+    tracked: set = set()
+
+    @callback
+    async def async_update_devices() -> None:
+        """Update the values of the devices."""
+        await async_add_tracked_entities(hass, tado, async_add_entities, tracked)
+
+    await async_update_devices()
+
+    async_dispatcher_connect(
+        hass,
+        "{DOMAIN}-update-{self.config_entry.entry_id}",
+        async_update_devices,
+    )
+
+
+async def async_add_tracked_entities(
+    hass: HomeAssistant,
+    tado: Any,
+    async_add_entities: AddEntitiesCallback,
+    tracked: set[str],
+) -> None:
+    """Add new tracker entities from Tado."""
+    _LOGGER.debug("Fetching Tado devices from API")
+    known_devices = await hass.async_add_executor_job(tado.get_me)
```

Comment:
We already have the known devices stored in the connector data.

Reply:
Fixed.